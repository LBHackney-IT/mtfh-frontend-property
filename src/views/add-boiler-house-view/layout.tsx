import React, { useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { AddBoilerHouseForm } from "../../components/add-boiler-house-form";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { ErrorSummary, Link, StatusBox } from "@mtfh/common/lib/components";

interface Props {
  asset: Asset;
}

export const AddBoilerHouseLayout = ({  asset }: Props) => {
  const [showSuccess, setShowSuccess] = useState(false);
  const [requestError, setRequestError] = useState<string | null>(null);

  return (
    <>
      <Link as={RouterLink} to={`/property/${asset.id}`} variant="back-link">
        Back to asset
      </Link>

      {showSuccess && (
        <StatusBox
          variant="success"
          title={
            <span>
              The boiler house has been added successfully.{" "}
              <Link as={RouterLink} to={`/property/${asset.id}`}>
                Return to property
              </Link>
            </span>
          }
        />
      )}

      <h1 className="lbh-heading-h1">Add boiler house to property</h1>
      <span className="govuk-caption-m lbh-caption">Some subtitle text</span>

      {requestError && (
        <ErrorSummary
          id="patch-asset-error"
          title="Unexpected error"
          description={requestError || undefined}
        />
      )}

      <AddBoilerHouseForm
        setShowSuccess={setShowSuccess}
        setRequestError={setRequestError}
        asset={asset}
      />
    </>
  );
};
