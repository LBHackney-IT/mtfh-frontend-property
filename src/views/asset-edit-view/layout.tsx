import React, { useEffect, useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { locale } from "../../services";

import { Address, getAddressViaUprn } from "@mtfh/common/lib/api/address/v1";
import { ErrorSummary, Link } from "@mtfh/common/lib/components";
import { EditableAddress } from "../../components/edit-asset-address-form/editable-address";
import { ReferenceAddress } from "../../components/edit-asset-address-form/reference-address";

import "./styles.scss";



export interface AssetEditLayoutProperties {
    assetDetails: Asset;
}


export const AssetEditLayout = ({ assetDetails }: AssetEditLayoutProperties): JSX.Element => {

    const [llpgAddress, setLlpgAddress] = useState<Address | null>(null);
    const [showError, setShowError] = useState<boolean>(false);
    const [errorMessage, setErrorMessage] = useState<string | null>(null);


    useEffect(() => {
        getAddressViaUprn(assetDetails.assetAddress.uprn).then((searchAddressResponse) => {
            if (searchAddressResponse.addresses) {
                setLlpgAddress(searchAddressResponse.addresses[0])
            }
        })
    }, []);

    return (
        <>
            <Link as={RouterLink} to={`/property/${assetDetails.id}`} variant="back-link">
                {locale.asset}
            </Link>
            <h1 className="lbh-heading-h1">Edit property address</h1>
            <span className="govuk-caption-m lbh-caption">New Addresses are suggested from the Local Gazetteer to bring some standardisation.</span>
            
            {showError && 
                  <ErrorSummary
                  id="patch-asset-error"
                  title={locale.errors.unableToPatchAsset}
                  description={errorMessage ? errorMessage : undefined}
                />
            }
            
            <div className="mtfh-address-details">
                <section><EditableAddress llpgAddress={llpgAddress} assetDetails={assetDetails} setShowError={setShowError} setErrorMessage={setErrorMessage}/></section>
                <section><ReferenceAddress assetAddressDetails={assetDetails.assetAddress} /></section>
            </div>
        </>
    );
};
