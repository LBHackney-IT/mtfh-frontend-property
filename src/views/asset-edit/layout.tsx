import React, { FC } from "react";
import { Link as RouterLink } from "react-router-dom";
import { Asset } from "@mtfh/common/lib/api/asset/v1";
import {
    Alert as AlertIcon,
    Button,
    Center,
    CommentList,
    Heading,
    Layout,
    Link,
    PageAnnouncement,
    PageAnnouncementProvider,
    SideBar,
    SideBarProps,
    Spinner,
    WorkOrderList,
} from "@mtfh/common/lib/components";


export interface AssetEditLayoutProperties {
    assetDetails: Asset;
}

export const AssetEditLayout: FC<AssetEditLayoutProperties> = ({ assetDetails }) => {


    return (
        <>
            <h3>Asset Edit View</h3>
            <p>{JSON.stringify(assetDetails)}</p>
        </>
        //   <PageAnnouncementProvider sessionKey="asset">
        //     <PageAnnouncement />
        //     <Layout
        //       backLink={
        //         <Link as={RouterLink} to="/search" variant="back-link">
        //           {locale.backToSearch}
        //         </Link>
        //       }
        //       top={
        //         <Heading variant="h1">
        //           {alertsData.alerts?.length > 0 && (
        //             <AlertIcon
        //               viewBox="0 0 37 58"
        //               width="28"
        //               height="44"
        //               style={{ margin: "-2px 4px 0 0" }}
        //             />
        //           )}
        //           {locale.assetDetails.address(assetDetails.assetAddress)}
        //         </Heading>
        //       }
        //       side={<AssetSideBar assetDetails={assetDetails} alerts={alertsData.alerts} />}
        //     >
        //       <PropertyBody assetId={assetDetails.assetId} propertyId={assetDetails.id} />
        //     </Layout>
        //   </PageAnnouncementProvider>
    );
};