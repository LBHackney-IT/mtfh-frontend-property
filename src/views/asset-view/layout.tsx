import React, { FC } from "react";
import { Link as RouterLink } from "react-router-dom";

import { AssetSideBar } from "../../components/asset-sidebar";
import { PropertyBody } from "../../components/property-body";
import { locale } from "../../services";
import { useCautionaryAlerts } from "./utils";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Alert as AlertIcon,
  Button,
  Center,
  Heading,
  Layout,
  Link,
  PageAnnouncement,
  PageAnnouncementProvider,
  Spinner,
} from "@mtfh/common/lib/components";

import "./styles.scss";

export interface Props {
  assetDetails: Asset;
  assetChildren: Asset[] | undefined;
  showTenureInformation: boolean;
  showCautionaryAlerts: boolean;
  enableNewProcesses: boolean;
}

export const AssetLayout: FC<Props> = ({
  assetDetails,
  assetChildren,
  showTenureInformation,
  showCautionaryAlerts,
  enableNewProcesses,
}) => {
  const { alertsData, isLoading: isLoadingCautionaryAlerts } = useCautionaryAlerts(
    assetDetails.assetId,
    showCautionaryAlerts,
  );

  const tenure = useTenure(assetDetails.tenure ? assetDetails.tenure.id : null).data;

  if (assetDetails.tenure && tenure && alertsData.length > 0) {
    const { householdMembers } = tenure;

    alertsData.forEach((alert) => {
      const matchingMember = householdMembers.find(
        (householdMember) =>
          alert.personName && alert.personName === householdMember.fullName,
      );

      if (matchingMember) {
        alert.personId = matchingMember.id;
      }
    });

  }


  if (isLoadingCautionaryAlerts) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  return (
    <PageAnnouncementProvider sessionKey="asset">
      <PageAnnouncement />
      <Layout
        backLink={
          <Link as={RouterLink} to="/search" variant="back-link">
            {locale.backToSearch}
          </Link>
        }
        top={
          <Heading variant="h1">
            {showCautionaryAlerts && alertsData.length > 0 && (
              <span data-test="cautionary-alerts-icon">
                <AlertIcon
                  viewBox="0 0 37 58"
                  width="28"
                  height="44"
                  style={{ margin: "-2px 4px 0 0" }}
                />
              </span>
            )}
            {locale.assets.assetDetails.address(assetDetails.assetAddress)}
          </Heading>
        }
        side={
          <AssetSideBar
            assetDetails={assetDetails}
            alerts={alertsData}
            showTenureInformation={showTenureInformation}
            showCautionaryAlerts={showCautionaryAlerts}
          />
        }
      >
        <PropertyBody
          assetDetails={assetDetails}
          childAssets={assetChildren}
          enableNewProcesses={enableNewProcesses}
          id={assetDetails.id}
        />
        
      </Layout>
    </PageAnnouncementProvider>
  );
};
