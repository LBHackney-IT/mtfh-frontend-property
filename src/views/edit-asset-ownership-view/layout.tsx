import React, { useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { EditAssetOwnershipForm } from "../../components/edit-asset-ownership-form";
import { locale } from "../../services";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { ErrorSummary, Link, StatusBox } from "@mtfh/common/lib/components";

interface EditAssetOwnershipLayoutProps {
  asset: Asset;
}

export const EditAssetOwnershipLayout = ({ asset }: EditAssetOwnershipLayoutProps) => {
  const [showSuccess, setShowSuccess] = useState(false);
  const [requestError, setRequestError] = useState<string | null>(null);

  return (
    <>
      <Link as={RouterLink} to={`/property/${asset.id}`} variant="back-link">
        Back to asset
      </Link>

      {showSuccess && (
        <StatusBox
          variant="success"
          data-testid="ownership-edit-success"
          // @ts-ignore No overload matches this call
          title={
            <span>
              {locale.lbhOwnershipInformation.editLbhOwnershipSuccessMessage}.{" "}
              <Link as={RouterLink} to={`/property/${asset.id}`}>
                View property
              </Link>
            </span>
          }
        />
      )}

      <h1 className="lbh-heading-h1">Edit asset ownership</h1>

      {requestError && (
        <ErrorSummary
          id="patch-asset-error"
          data-testid="ownership-edit-failure"
          title={locale.lbhOwnershipInformation.editOwnershipPatchError}
          description={locale.errors.tryAgainOrContactSupport}
        />
      )}

      <EditAssetOwnershipForm
        setShowSuccess={setShowSuccess}
        setRequestError={setRequestError}
        asset={asset}
      />
    </>
  );
};
