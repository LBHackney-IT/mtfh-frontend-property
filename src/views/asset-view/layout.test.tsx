import * as crypto from "crypto";

import React from "react";

import { render, server } from "@hackney/mtfh-test-utils";
import { waitFor } from "@testing-library/react";
import { rest } from "msw";

import { AssetLayout } from "./layout";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { Alert } from "@mtfh/common/lib/api/cautionary-alerts/v1/types";
import * as auth from "@mtfh/common/lib/auth/auth";
import { Patch } from "@mtfh/common/lib/api/patch/v1/types";

const assetData: Asset = {
  id: "769894bd-b0bc-47eb-a780-322372c2448f",
  patchId: crypto.randomBytes(20).toString("hex"),
  areaId: crypto.randomBytes(20).toString("hex"),
  assetId: "0019062023",
  assetType: "Block",
  assetLocation: {
    parentAssets: [],
    floorNo: "0",
    totalBlockFloors: 3,
  },
  assetAddress: {
    uprn: "10008355547",
    addressLine1: "123 Test Block(CAN BE DELETED)",
    addressLine2: "",
    addressLine3: "",
    addressLine4: "",
    postPreamble: "",
    postCode: "FK8 1FH",
  },
  assetManagement: {
    isCouncilProperty: true,
    managingOrganisation: "London Borough of Hackney",
    managingOrganisationId: "c01e3146-e630-c2cd-e709-18ef57bf3724",
    isTMOManaged: false,
    agent: "Agent Smith",
    areaOfficeName: "Test",
    owner: "Test",
  },
  assetCharacteristics: {
    numberOfBedrooms: 0,
    numberOfLifts: 0,
    numberOfLivingRooms: 0,
    windowType: "Wood",
    yearConstructed: "1991",
    numberOfSingleBeds: null,
    numberOfDoubleBeds: null,
    numberOfFloors: null,
    totalBlockFloors: null,
    heating: null,
    propertyFactor: null,
    architecturalType: null,
  },
  versionNumber: 0,
  tenure: {
    id: "c01e3146-e630-c2cd-e709-18ef57bf3724",
    paymentReference: "",
    type: "",
    startOfTenureDate: "",
    endOfTenureDate: "",
    isActive: false,
  },
  rootAsset: "",
  parentAssetIds: "",
  rentGroup: null,
  boilerHouseId: "",
};

const alert: Alert = {
  alertId: "1234",
  alertCode: "VA",
  assureReference: "",
  dateModified: "",
  description: "Verbal Abuse",
  endDate: null,
  modifiedBy: "",
  personName: "Joan Fisher",
  personId: "1",
  reason: "",
  startDate: "",
  isActive: true,
};

const mockPatch: Patch = {
  id: crypto.randomBytes(20).toString("hex"),
  name: "HN1",
  patchType: "patch",
  parentId: crypto.randomBytes(20).toString("hex"),
  domain: "Hackney",
  responsibleEntities: [
    {
      id: crypto.randomBytes(20).toString("hex"),
      name: "Housing Officer 1",
      responsibleType: "HousingOfficer",
      contactDetails: {
        emailAddress: "test.test@hackney.gov.uk",
      },
    },
  ],
};

beforeEach(() => {
  jest.resetAllMocks();

  jest.spyOn(auth, "isAuthorisedForGroups").mockReturnValue(true);

  server.use(
    rest.get(
      `/api/v1/cautionary-alerts/properties-new/${assetData.assetId}`,
      (req, res, ctx) => res(ctx.status(200), ctx.json({ alerts: [alert] })),
    ),
  );

  server.use(
      rest.get("/api/v1/patch/all", (req, res, ctx) => {
        return res(ctx.json([mockPatch]));
      }),
  );
  
});

test("it shows the new cautionary alerts icon", async () => {
  // Arrange
  const { container } = render(
    <AssetLayout
      assetDetails={assetData}
      assetChildren={[]}
      showTenureInformation={false}
      showCautionaryAlerts
      enableNewProcesses={false}
    />,
    {
      url: `/property/${assetData.id}`,
      path: "/property/:assetId",
    },
  );

  // Assert
  await waitFor(async () => {
    // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
    const cautionaryAlertsIcon = container.querySelector(
      "[data-test='cautionary-alerts-icon']",
    );

    expect(cautionaryAlertsIcon).toBeVisible();
  });

  expect(container).toMatchSnapshot();
});

test("it hides the cautionary alerts icon", () => {
  // Arrange
  const { container } = render(
    <AssetLayout
      assetDetails={assetData}
      assetChildren={[]}
      showTenureInformation={false}
      showCautionaryAlerts={false}
      enableNewProcesses={false}
    />,
    {
      url: `/property/${assetData.id}`,
      path: "/property/:assetId",
    },
  );

  // Assert
  // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
  const newProcessButton = container.querySelector(
    "[data-test='cautionary-alerts-icon']",
  );

  expect(newProcessButton).not.toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
