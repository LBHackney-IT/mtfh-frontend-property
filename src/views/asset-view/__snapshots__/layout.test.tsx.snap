// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it hides the cautionary alerts icon 1`] = `
<div>
  <div
    class="mtfh-layout"
  >
    <a
      class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
      href="/search"
    >
      Search results
    </a>
    <div
      id="content"
    />
    <h1
      class="lbh-heading-h1"
    >
      123 Test Block(CAN BE DELETED) FK8 1FH
    </h1>
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__aside"
      >
        <div
          class="mtfh-asset-sidebar"
        >
          <div
            class="mtfh-sidebar"
          >
            <div
              id="property-view-sidebar"
            >
              <aside
                class="mtfh-asset-details"
              >
                <dl
                  class="govuk-summary-list govuk-summary-list--no-border lbh-summary-list"
                >
                  <div
                    class="govuk-summary-list__row"
                  >
                    <dt
                      class="govuk-summary-list__key govuk-!-width-two-thirds"
                    >
                      Type
                    </dt>
                    <dd
                      class="govuk-summary-list__value"
                    >
                      Block
                    </dd>
                  </div>
                  <div
                    class="govuk-summary-list__row"
                  >
                    <dt
                      class="govuk-summary-list__key"
                    >
                      UPRN
                    </dt>
                    <dd
                      class="govuk-summary-list__value"
                    >
                      10008355547
                    </dd>
                  </div>
                  <div
                    class="govuk-summary-list__row"
                  >
                    <dt
                      class="govuk-summary-list__key"
                    >
                      Reference
                    </dt>
                    <dd
                      class="govuk-summary-list__value"
                    >
                      0019062023
                    </dd>
                  </div>
                </dl>
              </aside>
              <details
                class="govuk-details"
                data-module="govuk-details"
              >
                <summary
                  class="govuk-details__summary"
                  data-testid="property-spec-toggle"
                >
                  <span
                    class="govuk-details__summary-text"
                  >
                    Property Specification
                  </span>
                </summary>
              </details>
              <a
                class="govuk-button lbh-button"
                data-testid="edit-address-button"
                href="/property/edit/769894bd-b0bc-47eb-a780-322372c2448f"
              >
                Edit address details
              </a>
              <hr
                class="lbh-horizontal-bar"
              />
              <aside
                class="mtfh-patch-details"
              >
                <h2
                  class="lbh-heading-h2 lbh-heading lbh-heading-h3"
                >
                  Patch details
                </h2>
                <p>
                  No patch
                </p>
                <button
                  class="govuk-button lbh-button"
                  data-testid="edit-assignment-button"
                  style="margin-top: 0px;"
                  type="button"
                >
                  Edit
                </button>
                <a
                  class="govuk-button lbh-button"
                  data-testid="all-patches-and-areas-button"
                  href="/property/all-patches-and-areas"
                >
                  All patches and areas
                </a>
              </aside>
              <hr
                class="lbh-horizontal-bar"
              />
              <aside
                class="mtfh-lbh-ownership-info"
              >
                <h2
                  class="lbh-heading-h2 lbh-heading lbh-heading-h3"
                  data-testid="ownership-info-heading"
                >
                  Asset ownership
                </h2>
                <p
                  class="lbh-body-s"
                  data-testid="ownership-info"
                >
                  This asset is owned by Hackney Council
                </p>
                <a
                  class="govuk-button lbh-button"
                  data-testid="edit-ownership-info-button"
                  href="/property/769894bd-b0bc-47eb-a780-322372c2448f/edit-asset-ownership"
                >
                  Edit ownership
                </a>
                <hr
                  class="lbh-horizontal-bar"
                />
              </aside>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mtfh-layout__main"
      >
        <div
          id="property-body-grid-container"
        >
          <div
            id="property-tree-grid-area"
          >
            <div>
              <div
                class="rst__tree"
              >
                <div
                  class="rst__node"
                  style="height: 62px;"
                >
                  <div
                    class="rst__lineBlock rst__lineHalfHorizontalRight"
                    style="width: 44px;"
                  />
                  <div
                    class="rst__nodeContent"
                    style="left: 44px;"
                  >
                    <div
                      style="height: 100%;"
                    >
                      <div>
                        <button
                          aria-label="Collapse"
                          class="rst__collapseButton"
                          style="left: -22px;"
                          type="button"
                        />
                        <div
                          class="rst__lineChildren"
                          style="width: 44px;"
                        />
                      </div>
                      <div
                        class="rst__rowWrapper"
                      >
                        <div
                          class="rst__row"
                          style="opacity: 1;"
                        >
                          <div
                            class="rst__moveHandle"
                            draggable="true"
                          />
                          <div
                            class="rst__rowContents"
                          >
                            <div
                              class="rst__rowLabel"
                            >
                              <span
                                class="rst__rowTitle"
                              >
                                <span>
                                  Hackney
                                </span>
                              </span>
                            </div>
                            <div
                              class="rst__rowToolbar"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="rst__node"
                  style="height: 62px;"
                >
                  <div
                    class="rst__lineBlock"
                    style="width: 44px;"
                  />
                  <div
                    class="rst__lineBlock rst__lineHalfVerticalTop rst__lineHalfHorizontalRight"
                    style="width: 44px;"
                  />
                  <div
                    class="rst__nodeContent"
                    style="left: 88px;"
                  >
                    <div
                      style="height: 100%;"
                    >
                      <div
                        class="rst__rowWrapper"
                      >
                        <div
                          class="rst__row"
                          style="opacity: 1;"
                        >
                          <div
                            class="rst__moveHandle"
                            draggable="true"
                          />
                          <div
                            class="rst__rowContents"
                          >
                            <div
                              class="rst__rowLabel"
                            >
                              <span
                                class="rst__rowTitle"
                              >
                                123 Test Block(CAN BE DELETED) (this asset)
                              </span>
                            </div>
                            <div
                              class="rst__rowToolbar"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            id="repairs-grid-area"
          />
          <div
            id="comments-grid-area"
          >
            <h2
              class="lbh-heading-h2"
            >
              Comments
            </h2>
            <a
              class="govuk-button lbh-button"
              href="/comment/property/769894bd-b0bc-47eb-a780-322372c2448f"
            >
              Add comment
            </a>
            <div>
              <div
                class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
              >
                <svg
                  class="mtfh-icon"
                  focusable="false"
                  height="50"
                  stroke="#00703c"
                  viewBox="0 0 42 42"
                  width="50"
                >
                  <title>
                    Loading...
                  </title>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <g
                      stroke-width="5"
                      transform="translate(3 3)"
                    >
                      <circle
                        cx="18"
                        cy="18"
                        r="18"
                        stroke-opacity=".5"
                      />
                      <path
                        d="M36 18c0-9.94-8.06-18-18-18"
                        transform="rotate(112.708 18 18)"
                      >
                        <animatetransform
                          attributeName="transform"
                          dur="1s"
                          from="0 18 18"
                          repeatCount="indefinite"
                          to="360 18 18"
                          type="rotate"
                        />
                      </path>
                    </g>
                  </g>
                </svg>
              </div>
            </div>
          </div>
          <div>
            <a
              class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
              href="/activities/property/769894bd-b0bc-47eb-a780-322372c2448f"
            >
              Activity history
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it shows the new cautionary alerts icon 1`] = `
<div>
  <div
    class="mtfh-layout"
  >
    <a
      class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
      href="/search"
    >
      Search results
    </a>
    <div
      id="content"
    />
    <h1
      class="lbh-heading-h1"
    >
      <span
        data-test="cautionary-alerts-icon"
      >
        <svg
          class="mtfh-icon"
          focusable="false"
          height="44"
          style="margin: -2px 4px 0px 0px;"
          viewBox="0 0 37 58"
          width="28"
        >
          <path
            d="M16.5 42.6212C18.7688 42.6212 20.625 40.7605 20.625 38.4863H12.375C12.375 40.7605 14.2312 42.6212 16.5 42.6212Z"
            fill="#BE3A34"
          />
          <path
            d="M28.875 30.2225V19.8849C28.875 13.5376 25.5131 8.22412 19.5938 6.81821V5.4123C19.5938 3.69627 18.2119 2.31104 16.5 2.31104C14.7881 2.31104 13.4062 3.69627 13.4062 5.4123V6.81821C7.5075 8.22412 4.125 13.517 4.125 19.8849V30.2225L0 34.3575V36.425H33V34.3575L31.1268 32.4797L28.875 30.2225Z"
            fill="#BE3A34"
          />
          <path
            d="M17.5808 25.7373H15.4363L14.988 17.1504H18.0291L17.5808 25.7373ZM14.9529 28.7432C14.9529 28.1924 15.1023 27.8057 15.4011 27.583C15.7058 27.3604 16.072 27.249 16.4998 27.249C16.9158 27.249 17.2732 27.3604 17.572 27.583C17.8767 27.8057 18.0291 28.1924 18.0291 28.7432C18.0291 29.2705 17.8767 29.6514 17.572 29.8857C17.2732 30.1201 16.9158 30.2373 16.4998 30.2373C16.072 30.2373 15.7058 30.1201 15.4011 29.8857C15.1023 29.6514 14.9529 29.2705 14.9529 28.7432Z"
            fill="white"
          />
        </svg>
      </span>
      123 Test Block(CAN BE DELETED) FK8 1FH
    </h1>
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__aside"
      >
        <div
          class="mtfh-asset-sidebar"
        >
          <div
            class="mtfh-sidebar"
          >
            <div
              id="property-view-sidebar"
            >
              <aside
                class="mtfh-asset-details"
              >
                <dl
                  class="govuk-summary-list govuk-summary-list--no-border lbh-summary-list"
                >
                  <div
                    class="govuk-summary-list__row"
                  >
                    <dt
                      class="govuk-summary-list__key govuk-!-width-two-thirds"
                    >
                      Type
                    </dt>
                    <dd
                      class="govuk-summary-list__value"
                    >
                      Block
                    </dd>
                  </div>
                  <div
                    class="govuk-summary-list__row"
                  >
                    <dt
                      class="govuk-summary-list__key"
                    >
                      UPRN
                    </dt>
                    <dd
                      class="govuk-summary-list__value"
                    >
                      10008355547
                    </dd>
                  </div>
                  <div
                    class="govuk-summary-list__row"
                  >
                    <dt
                      class="govuk-summary-list__key"
                    >
                      Reference
                    </dt>
                    <dd
                      class="govuk-summary-list__value"
                    >
                      0019062023
                    </dd>
                  </div>
                </dl>
              </aside>
              <details
                class="govuk-details"
                data-module="govuk-details"
              >
                <summary
                  class="govuk-details__summary"
                  data-testid="property-spec-toggle"
                >
                  <span
                    class="govuk-details__summary-text"
                  >
                    Property Specification
                  </span>
                </summary>
              </details>
              <a
                class="govuk-button lbh-button"
                data-testid="edit-address-button"
                href="/property/edit/769894bd-b0bc-47eb-a780-322372c2448f"
              >
                Edit address details
              </a>
              <hr
                class="lbh-horizontal-bar"
              />
              <aside
                class="mtfh-cautionary-alerts"
              >
                <h2
                  class="lbh-heading-h2 lbh-heading lbh-heading-h3"
                >
                  <svg
                    class="mtfh-icon"
                    data-testid="alert-icon"
                    focusable="false"
                    height="26"
                    style="margin-right: 4px;"
                    viewBox="0 0 37 58"
                    width="16"
                  >
                    <path
                      d="M16.5 42.6212C18.7688 42.6212 20.625 40.7605 20.625 38.4863H12.375C12.375 40.7605 14.2312 42.6212 16.5 42.6212Z"
                      fill="#BE3A34"
                    />
                    <path
                      d="M28.875 30.2225V19.8849C28.875 13.5376 25.5131 8.22412 19.5938 6.81821V5.4123C19.5938 3.69627 18.2119 2.31104 16.5 2.31104C14.7881 2.31104 13.4062 3.69627 13.4062 5.4123V6.81821C7.5075 8.22412 4.125 13.517 4.125 19.8849V30.2225L0 34.3575V36.425H33V34.3575L31.1268 32.4797L28.875 30.2225Z"
                      fill="#BE3A34"
                    />
                    <path
                      d="M17.5808 25.7373H15.4363L14.988 17.1504H18.0291L17.5808 25.7373ZM14.9529 28.7432C14.9529 28.1924 15.1023 27.8057 15.4011 27.583C15.7058 27.3604 16.072 27.249 16.4998 27.249C16.9158 27.249 17.2732 27.3604 17.572 27.583C17.8767 27.8057 18.0291 28.1924 18.0291 28.7432C18.0291 29.2705 17.8767 29.6514 17.572 29.8857C17.2732 30.1201 16.9158 30.2373 16.4998 30.2373C16.072 30.2373 15.7058 30.1201 15.4011 29.8857C15.1023 29.6514 14.9529 29.2705 14.9529 28.7432Z"
                      fill="white"
                    />
                  </svg>
                  Cautionary alerts
                </h2>
                <p
                  class="lbh-body-s"
                >
                  <a
                    class="govuk-link lbh-link lbh-link--no-visited-state person-link"
                    href="/person/1"
                  >
                    Joan Fisher
                  </a>
                  <br />
                  Verbal Abuse
                   
                  <br />
                </p>
                <hr
                  class="lbh-horizontal-bar"
                />
              </aside>
              <aside
                class="mtfh-patch-details"
              >
                <h2
                  class="lbh-heading-h2 lbh-heading lbh-heading-h3"
                >
                  Patch details
                </h2>
                <p>
                  No patch
                </p>
                <button
                  class="govuk-button lbh-button"
                  data-testid="edit-assignment-button"
                  style="margin-top: 0px;"
                  type="button"
                >
                  Edit
                </button>
                <a
                  class="govuk-button lbh-button"
                  data-testid="all-patches-and-areas-button"
                  href="/property/all-patches-and-areas"
                >
                  All patches and areas
                </a>
              </aside>
              <hr
                class="lbh-horizontal-bar"
              />
              <aside
                class="mtfh-lbh-ownership-info"
              >
                <h2
                  class="lbh-heading-h2 lbh-heading lbh-heading-h3"
                  data-testid="ownership-info-heading"
                >
                  Asset ownership
                </h2>
                <p
                  class="lbh-body-s"
                  data-testid="ownership-info"
                >
                  This asset is owned by Hackney Council
                </p>
                <a
                  class="govuk-button lbh-button"
                  data-testid="edit-ownership-info-button"
                  href="/property/769894bd-b0bc-47eb-a780-322372c2448f/edit-asset-ownership"
                >
                  Edit ownership
                </a>
                <hr
                  class="lbh-horizontal-bar"
                />
              </aside>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mtfh-layout__main"
      >
        <div
          id="property-body-grid-container"
        >
          <div
            id="property-tree-grid-area"
          >
            <div>
              <div
                class="rst__tree"
              >
                <div
                  class="rst__node"
                  style="height: 62px;"
                >
                  <div
                    class="rst__lineBlock rst__lineHalfHorizontalRight"
                    style="width: 44px;"
                  />
                  <div
                    class="rst__nodeContent"
                    style="left: 44px;"
                  >
                    <div
                      style="height: 100%;"
                    >
                      <div>
                        <button
                          aria-label="Collapse"
                          class="rst__collapseButton"
                          style="left: -22px;"
                          type="button"
                        />
                        <div
                          class="rst__lineChildren"
                          style="width: 44px;"
                        />
                      </div>
                      <div
                        class="rst__rowWrapper"
                      >
                        <div
                          class="rst__row"
                          style="opacity: 1;"
                        >
                          <div
                            class="rst__moveHandle"
                            draggable="true"
                          />
                          <div
                            class="rst__rowContents"
                          >
                            <div
                              class="rst__rowLabel"
                            >
                              <span
                                class="rst__rowTitle"
                              >
                                <span>
                                  Hackney
                                </span>
                              </span>
                            </div>
                            <div
                              class="rst__rowToolbar"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="rst__node"
                  style="height: 62px;"
                >
                  <div
                    class="rst__lineBlock"
                    style="width: 44px;"
                  />
                  <div
                    class="rst__lineBlock rst__lineHalfVerticalTop rst__lineHalfHorizontalRight"
                    style="width: 44px;"
                  />
                  <div
                    class="rst__nodeContent"
                    style="left: 88px;"
                  >
                    <div
                      style="height: 100%;"
                    >
                      <div
                        class="rst__rowWrapper"
                      >
                        <div
                          class="rst__row"
                          style="opacity: 1;"
                        >
                          <div
                            class="rst__moveHandle"
                            draggable="true"
                          />
                          <div
                            class="rst__rowContents"
                          >
                            <div
                              class="rst__rowLabel"
                            >
                              <span
                                class="rst__rowTitle"
                              >
                                123 Test Block(CAN BE DELETED) (this asset)
                              </span>
                            </div>
                            <div
                              class="rst__rowToolbar"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            id="repairs-grid-area"
          />
          <div
            id="comments-grid-area"
          >
            <h2
              class="lbh-heading-h2"
            >
              Comments
            </h2>
            <a
              class="govuk-button lbh-button"
              href="/comment/property/769894bd-b0bc-47eb-a780-322372c2448f"
            >
              Add comment
            </a>
            <div>
              <div
                class="mtfh-center mtfh-center--horizontal mtfh-center--vertical"
              >
                <svg
                  class="mtfh-icon"
                  focusable="false"
                  height="50"
                  stroke="#00703c"
                  viewBox="0 0 42 42"
                  width="50"
                >
                  <title>
                    Loading...
                  </title>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <g
                      stroke-width="5"
                      transform="translate(3 3)"
                    >
                      <circle
                        cx="18"
                        cy="18"
                        r="18"
                        stroke-opacity=".5"
                      />
                      <path
                        d="M36 18c0-9.94-8.06-18-18-18"
                        transform="rotate(112.708 18 18)"
                      >
                        <animatetransform
                          attributeName="transform"
                          dur="1s"
                          from="0 18 18"
                          repeatCount="indefinite"
                          to="360 18 18"
                          type="rotate"
                        />
                      </path>
                    </g>
                  </g>
                </svg>
              </div>
            </div>
          </div>
          <div>
            <a
              class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
              href="/activities/property/769894bd-b0bc-47eb-a780-322372c2448f"
            >
              Activity history
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
